proposals/1125_dynamic-supervisor-oidc-clients/README.md

```diff
@@ -0,0 +1,507 @@
+  a running Supervisor by the admin.
+
+## Proposal
+
+### Goals and Non-goals
+
+Goals for this proposal:
+
+- Allow Pinniped admins to configure applications (OIDC clients) other than the Pinniped CLI to interact with the
+  Supervisor.
+- Provide a mechanism which governs a client's access to the token exchange APIs. Not all webapps should have permission
+  to act on behalf of the user with the Kubernetes API of the clusters, so an admin must be able to configure which
+  clients have this permission.
+- Provide a mechanism for requesting access to different aspects of a user identity, especially getting group
+  memberships or not, to allow the admin to exclude this potentially information for clients which do not need it.
+- Support a web UI based LDAP/ActiveDirectory login screen. This is needed to avoid having webapps handle the user's
+  password, which must only be seen by the Supervisor and the LDAP server. However, the details of this item have been
+  split out to a separate proposal document.
+- Client secrets must be stored encrypted or hashed, not in plain text.
+- Creation of client credentials on the operator's behalf - the server must generate any secrets.
+- The operator must be able to initiate manual rotation of client credentials.
+- Documentation describing the token exchanges a webapp backend must perform to interact with the Kubernetes API.
+
+Non-goals for this proposal:
+
+- Pinniped's scope is to provide authentication for cluster users. Providing authentication for arbitrary users to
+  arbitrary webapps is out of scope. The only proposed use case is providing the exact same identities that are provided
+  by using Pinniped's `kubectl` integration, which are the developers/devops/admin users of the cluster.
+- Supporting any OAuth/OIDC flow other
+  than [OIDC authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
+- Implementing any self-service client registration API. Clients will be registered by the Pinniped admin user.
+- Implementing a consent screen. This would be clearly valuable but will be left as a potential future enhancement in
+  the interest of keeping the first draft of this feature smaller.
+- Orchestration of token exchanges on behalf of the client. Webapps which want to make calls to the Kubernetes API of
+  clusters acting as the authenticated user will need to perform the rest of the token and credential exchange flow that
+  it currently implemented by the Pinniped CLI. Providing some kind of component or library to assist webapp developers
+  with these steps might be valuable but will be left as a potential future enhancement.
+- Supporting JWT-based client auth as described in [RFC 7523](https://datatracker.ietf.org/doc/html/rfc7523).  For now
+  client secret basic auth will be used.  This is left as a potential future enhancement.
+- Supporting public clients.
```

Comment:
In this context, "public" clients refers to clients that are not required to authenticate in authcode flows. E.g. the Pinniped CLI uses a public OIDC client, since there would be no point in hardcoding a client secret into the CLI client code, thus rendering it a non-secret.

Reply:
OIC, yes, that was not clear to me at all, but makes complete sense with the extra context.